IdentityCore/src/controllers/broker/mac/MSIDXpcSilentTokenRequestController.m

```diff
@@ -55,4 +79,24 @@ + (BOOL)canPerformRequest
     }
 }
 
+- (void)shouldSkipAcquireTokenBasedOn:(NSError *)error
+{
+    self.shouldSkipAcquireToken = YES;
+    MSID_LOG_WITH_CTX(MSIDLogLevelInfo, self.requestParameters, @"Looking if we should fallback to Xpc controller, error: %ld error domain: %@.", (long)error.code, error.domain);
+    // If it is MSIDErrorDomain and Sso Extension returns unexpected error, we should fall back to local controler and unblock user
+    if (![error.domain isEqualToString:ASAuthorizationErrorDomain] && ![error.domain isEqualToString:MSIDErrorDomain]) {
+    }
+    
+    switch (error.code)
+    {
+        case ASAuthorizationErrorNotHandled:
+        case ASAuthorizationErrorUnknown:
+        case ASAuthorizationErrorFailed:
+        case MSIDErrorSSOExtensionUnexpectedError:
+            self.shouldSkipAcquireToken = NO;
```

Comment:
Can the controller be reused for multiple requests? If it is, controller's shouldSkipAcquireToken might get set to NO in 1st request and remain NO for second request